app/api/currentuser/update_notification_read_date.go

```diff
@@ -0,0 +1,41 @@
+package currentuser
+
+// swagger:operation PUT /current-user/notification-read-date userNotificationReadDateUpdate
+// ---
+// summary: Update user's notification read date
+// description: Set users.sNotificationReadDate to NOW() for the current user
+// responses:
+//   "200":
+//     "$ref": "#/responses/updatedResponse"
+//   "401":
+//     "$ref": "#/responses/unauthorizedResponse"
+//   "403":
+//     "$ref": "#/responses/forbiddenResponse"
+//   "500":
+//     "$ref": "#/responses/internalErrorResponse"
+
+import (
+	"net/http"
+
+	"github.com/go-chi/render"
+	"github.com/jinzhu/gorm"
+
+	"github.com/France-ioi/AlgoreaBackend/app/database"
+	"github.com/France-ioi/AlgoreaBackend/app/service"
+)
+
+func (srv *Service) updateNotificationReadDate(w http.ResponseWriter, r *http.Request) service.APIError {
+	user := srv.GetUser(r)
+	err := user.Load()
+	if err == database.ErrUserNotFound {
```

Comment:
I would not do `InsufficientAccessRightsError` as it is not really a problem of "forbidden" and normally if his token was accepted (in the middleware), it exists.  
Actually, you could do it by checking the row affected by the update instead (no extra query).
But I think I would let the update flow silently in any case. Tell me if you disagree.
(if you do so, don't forget to remove "403" from the possible responses in the doc.)

Reply:
fixed